cmd/cue/cmd/get_go.go

```diff
@@ -473,7 +486,25 @@ func (e *extractor) extractPkg(root string, p *packages.Package) error {
 		args += " --exclude=" + e.exclude
 	}
 
+	pName := flagPackage.String(e.cmd)
+	if pName == "" {
+		pName = p.Name
+	}
+
+	var (
+		decls  []cueast.Decl
+		cuePkg *cueast.Package
+	)
 	for i, f := range p.Syntax {
+		// If we are in the first iteration we need to set cuePkg.
+		// Af we want to join everything into the same file,
+		// we have to keep the same cuePkg and declarations around
+		// for all iterations, else reset.
```

Comment:
I'd move this comment above the for loop, and also explain what's happening with writing only the last (joined) file. Something like...

```
// By default, we output one CUE file for each Go file as long as there's anything to generate.
// When --outfile is used, we want to generate exactly one CUE file for an entire Go package,
// so we instead keep joining declarations until we reach the last Go file, where we then write.
```